apis/v1alpha1/targetallocator_types.go

```diff
@@ -87,7 +87,7 @@ type TargetAllocatorSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Observability"
 	Observability v1beta1.ObservabilitySpec `json:"observability,omitempty"`
 	// CollectorNotReadyGracePeriod defines the grace period after which a TargetAllocator stops considering a collector is target assignable.
-	// The default is 0s, which means that all collectors can be assigned targets irrespective of their readiness.
+	// The default is 30s, which means collectors will only be assigned targets if they have been ready for at least 30s.
```

Comment:
This description isn't right. What this setting does is allow a collector which becomes not-Ready to still have targets assigned to it for 30 seconds, with the assumption that its state is temporary.

```suggestion
	// The default is 30s, which means that if a collector becomes not Ready, the target allocator will wait for 30 seconds before reassigning its targets. The assumption is that the state is temporary, and an expensive target reallocation should be avoided if possible.
```